crates/cli/cli/src/commands/query/mod.rs

```diff
@@ -57,7 +54,7 @@ impl CommandRunner<HermesApp> for QueryCommands {
         match self {
             Self::Client(cmd) => app.run_command(cmd).await,
             Self::Clients(cmd) => cmd.run(app).await,
-            Self::Connection(cmd) => cmd.run(app).await,
+            Self::Connection(cmd) => app.run_command(cmd).await,
```

Comment:
We need to add the following wiring to [`HermesCommandRunnerComponents`](https://github.com/informalsystems/hermes-sdk/blob/668e55080bef8e13ffffbeceba49d5be74de0ff8/crates/cli/cli/src/contexts/app.rs#L161) to forward the subcommand to the corresponding runners:

```
        QueryConnectionSubCommand: RunQueryConnectionSubCommand,
        QueryConnectionEndArgs: RunQueryConnectionEndCommand,
```

Reply:
@soareschen Im getting error 
```shell
error[E0277]: the trait bound `RunQueryConnectionSubCommand: hermes_cli_components::traits::command::CommandRunner<HermesApp, QueryConnectionSubCommand>` is not satisfied
  --> crates/cli/cli/src/commands/query/mod.rs:58:54
   |
58 |             Self::Connection(cmd) => app.run_command(cmd).await,
   |                                          ----------- ^^^ the trait `hermes_cli_components::traits::command::CommandRunner<HermesApp, QueryConnectionSubCommand>` is not implemented for `RunQueryConnectionSubCommand`, which is required by `HermesApp: CanRunCommand<_>`
   |                                          |
   |                                          required by a bound introduced by this call
   |
   = help: the trait `hermes_cli_components::traits::command::CommandRunner<App, QueryConnectionSubCommand>` is implemented for `RunQueryConnectionSubCommand`
   = note: required for `UseDelegate<HermesCommandRunnerComponents>` to implement `hermes_cli_components::traits::command::CommandRunner<HermesApp, QueryConnectionSubCommand>`
   = note: 1 redundant requirement hidden
   = note: required for `HermesAppComponents` to implement `hermes_cli_components::traits::command::CommandRunner<HermesApp, QueryConnectionSubCommand>`
   = note: required for `HermesApp` to implement `CanRunCommand<QueryConnectionSubCommand>`

error[E0277]: the trait bound `UseDelegate<HermesCommandRunnerComponents>: hermes_cli_components::traits::command::CommandRunner<HermesApp, QueryConnectionSubCommand>` is not satisfied
  --> crates/cli/cli/src/commands/query/mod.rs:58:38
   |
58 |             Self::Connection(cmd) => app.run_command(cmd).await,
   |                                      ^^^^^^^^^^^^^^^^^^^^ the trait `hermes_cli_components::traits::command::CommandRunner<HermesApp, QueryConnectionSubCommand>` is not implemented for `UseDelegate<HermesCommandRunnerComponents>`, which is required by `HermesAppComponents: hermes_cli_components::traits::command::CommandRunner<HermesApp, QueryConnectionSubCommand>`
   |
   = help: the trait `hermes_cli_components::traits::command::CommandRunner<App, Args>` is implemented for `UseDelegate<Components>`
   = note: required for `HermesAppComponents` to implement `hermes_cli_components::traits::command::CommandRunner<HermesApp, QueryConnectionSubCommand>`
note: required by a bound in `<App as CanRunCommand<Args>>`
  --> /home/thaodinh/projects/oss/informalsystems/hermes-sdk/crates/cli/cli-components/src/traits/command.rs:6:1
   |
6  | #[derive_component(CommandRunnerComponent, CommandRunner<App>)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `<App as CanRunCommand<Args>>`
   = note: this error originates in the attribute macro `derive_component` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `UseDelegate<HermesCommandRunnerComponents>: hermes_cli_components::traits::command::CommandRunner<HermesApp, QueryConnectionSubCommand>` is not satisfied
  --> crates/cli/cli/src/commands/query/mod.rs:58:59
   |
58 |             Self::Connection(cmd) => app.run_command(cmd).await,
   |                                                           ^^^^^ the trait `hermes_cli_components::traits::command::CommandRunner<HermesApp, QueryConnectionSubCommand>` is not implemented for `UseDelegate<HermesCommandRunnerComponents>`, which is required by `HermesAppComponents: hermes_cli_components::traits::command::CommandRunner<HermesApp, QueryConnectionSubCommand>`
   |
   = help: the trait `hermes_cli_components::traits::command::CommandRunner<App, Args>` is implemented for `UseDelegate<Components>`
   = note: required for `HermesAppComponents` to implement `hermes_cli_components::traits::command::CommandRunner<HermesApp, QueryConnectionSubCommand>`
 ```
while I already added all necessary constraints, also I tried to expand macros to check the code generated by cgp, but I cannot find out what actually the problem is. Could you please shed some light?

Reply:
thank for your guidance @soareschen to help me fix this.